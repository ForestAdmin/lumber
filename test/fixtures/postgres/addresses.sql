CREATE TABLE addresses (
  user_id INT NOT NULL,
  street VARCHAR(30) NOT NULL,
  city VARCHAR(30) NOT NULL CONSTRAINT unique_city UNIQUE,
  state VARCHAR(30) NOT NULL,
  PRIMARY KEY (user_id),
  CONSTRAINT fk_user_id FOREIGN KEY (user_id) REFERENCES users (id)
);
