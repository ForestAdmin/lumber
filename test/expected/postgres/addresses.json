{
  "fields": [
    {
      "name": "id",
      "nameColumn": "id",
      "type": "UUID",
      "primaryKey": true,
      "defaultValue": null
    }
  ],
  "references": [
    {
      "ref": "customers",
      "foreignKey": "customer_id",
      "foreignKeyName": "customerId",
      "as": "customer"
    },{
      "as": "city",
      "foreignKey": "city",
      "foreignKeyName": "cityKey",
      "ref": "addresses",
      "targetKey": "city"
    }
  ],
  "primaryKeys": [
    "id"
  ],
  "options": {
    "underscored": false,
    "timestamps": false,
    "hasIdColumn": true,
    "hasPrimaryKeys": true
  }
}
